DOCS-3211 Document new + Add a backup credential button (#1290)
truenas/documentation: content/SCALE/DataProtection/Cloud Sync Tasks/_index.md

{{< toc >}}

TrueNAS can send, receive, or synchronize data with a cloud storage provider. Cloud sync tasks allow for single-time transfers or recurring transfers on a schedule. They are an effective method to back up data to a remote location.

{{< hint warning >}}

> Using the cloud means data can go to a third-party commercial vendor not directly affiliated with iXsystems. You should fully understand vendor pricing policies and services before using them for cloud sync tasks. iXsystems is not responsible for any charges incurred from using third-party vendors with the Cloud Sync feature.

Reviewer: ```suggestion to draw more attention to it, placed last line in separate paragraph
Using the cloud means data can go to a third-party commercial vendor not directly affiliated with iXsystems. You should fully understand vendor pricing policies and services before using them for cloud sync tasks.

 iXsystems is not responsible for any charges incurred from using third-party vendors with the cloud sync feature.
```

{{< /hint >}}

> TrueNAS supports major providers like Amazon S3, Google Cloud, and Microsoft Azure. It also supports many other vendors. To see the full list of supported vendors, go to **Credentials > Backup Credentials > Cloud Credentials** click **Add** and open the **Provider** dropdown.

Reviewer: ```suggestion
TrueNAS supports major providers like Amazon S3, Google Cloud, and Microsoft Azure. It also supports many other vendors. To see the full list of supported vendors, go to **Credentials > Backup Credentials > Cloud Credentials** click **Add** and open the **Provider** drop-down list.
```

## Requirements

> * You must have all system [Storage](/scale/storage/) configured and ready to receive or send data.

Reviewer: ```suggestion
* You must have all system [storage](/scale/storage/) configured and ready to receive or send data.
```

Note that syncing to a Backblaze B2 bucket does not delete files from the bucket, even when you deleted those files locally. Instead, files are tagged with a version number or moved to a hidden state. To automatically delete old or unwanted files from the bucket, adjust the [Backblaze B2 Lifecycle Rules](https://www.backblaze.com/blog/backblaze-b2-lifecycle-rules/).

Sync cannot delete files stored in Amazon S3 Glacier or S3 Glacier Deep Archive. These files must first be restored by another means, like the [Amazon S3 console](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/restore-archived-objects.html).
{{< /tab >}}
{{< tab "Copy" >}}
**Copy** duplicates each source file into the destination, overwriting any destination files using the same name as the source. Copying is the least potentially destructive option.
{{< /tab >}}
{{< tab "Move" >}}
**Move** transfers the files from the source to the destination and deletes the source files. **Move** also overwrites files with the same names on the destination.
{{< /tab >}}
{{< /tabs >}}

> Next, Use the **Control** options. Define when the task runs using **Schedule**. If you need a specific schedule, select **Custom** and use the **Advanced Scheduler**.

Reviewer: ```suggestion
Next, use the **Control** options. Define when the task runs using **Schedule**. If you need a specific schedule, select **Custom** and use the **Advanced Scheduler**.
```

{{< expand "Advanced Scheduler" "v" >}}
{{< include file="static/includes/SCALE/SCALEAdvancedScheduler.md.part" markdown="true" >}}
{{< /expand >}}

> Uncheck the **Enable** checkbox to make the configuration available without allowing the specified schedule to run the task. To manually activate a saved task, go to **Data Protection > Cloud Sync Tasks**, click <i class="fa fa-play" aria-hidden="true"></i> for the cloud sync task you want to run.  You are prompted to **CONTINUE** or **CANCEL** the **Run Now** operation.

Reviewer: ```suggestion
Clear the **Enable** checkbox to make the configuration available without allowing the specified schedule to run the task. To manually activate a saved task, go to **Data Protection > Cloud Sync Tasks**, click <i class="fa fa-play" aria-hidden="true"></i> for the cloud sync task you want to run.  You are prompted to **CONTINUE** or **CANCEL** the **Run Now** operation.
```

### Scripting and Environment Variables

> Advanced users can write scripts that run immediately *before* or *after* the Cloud Sync task. The **Post-script** field only runs when the cloud sync task succeeds. You can pass a variety of task environment variables into the **Pre-** and **Post-** script fields:

Reviewer: ```suggestion
Advanced users can write scripts that run immediately before or after the cloud sync task. The **Post-script** field only runs when the cloud sync task succeeds. You can pass a variety of task environment variables into the **Pre-** and **Post-** script fields:
```

### Testing Settings

To test the settings before saving, click **Dry Run**. TrueNAS connects to the cloud storage provider and simulates a file transfer but does not send or receive data.

> A dialog box displays with the test status and allows you to download the task logs. You can also run this by clicking <i class="material-icons" aria-hidden="true" title="Dry Run">loop</i> after the task is created.

Reviewer: ```suggestion
A dialog box displays with the test status and allows you to download the task logs. You can also run this by clicking <i class="material-icons" aria-hidden="true" title="Dry Run">loop</i> after creating the task.
```

Saved tasks activate according to their schedule or by clicking <i class="material-icons" aria-hidden="true" title="Run Now">play_arrow</i>. An in-progress cloud sync must finish before another can begin. Stopping an in-progress task cancels the file transfer and requires starting the file transfer over.

> To view logs about a running task, or its most recent run, click *State*.

Reviewer: ```suggestion
To view logs about a running task, or its most recent run, click **State**.
```
